lib/payload.json

```diff
@@ -34,6 +51,40 @@
           "format": "date-time",
           "description": "Date and time of the measurement (RFC3339)"
         },
+        "soil": {
+          "type": "object",
+          "properties": {
+            "moisture": {
+              "description": "Soil moisture (%)",
+              "$ref": "#/definitions/percentage"
+            },
+            "temperature": {
+              "description": "Soil temperature (°C)",
+              "$ref": "#/definitions/temperature"
+            },
+            "EC": {
+              "description": "Soil electrical conductivity (dS/m)",
+              "$ref": "#/definitions/conductivity"
+            },
+            "pH": {
+              "description": "Soil pH level (open-ended scale)",
+              "type": "number"
+            },
```

Comment:
I would say that `[0..14]` is a sensible practical range, indeed for the environment we're aiming at.

If we define validity (ranges, patterns) I think we should extract it to its own definition.